libraries/reliable_amqp_publisher/reliable_amqp_publisher.cpp

```diff
@@ -40,7 +43,13 @@ struct reliable_amqp_publisher_impl {
    std::atomic_bool stopping = false;
 
    unsigned in_flight = 0;
-   std::deque<std::vector<char>> message_deque;
+   fc::unsigned_int batch_num = 0;
```

Comment:
My use case is per block so uint32_t would be plenty as that is same as block_num. However, I guess if used for something with higher rate, it could be an issue. I'll handle overflow to reset to 1.